### ISSUE TITLE
"docker login" doesn't run in Linux without X11

### ISSUE THREAD

**Issue by SundialServices** [docker/cli#2229]
I cannot use the command `docker login quay.io` in Linux unless I am running in a command-line environment that is hosted by some kind of GUI.  Otherwise, I get: *(whether or not it's "quay.io" ...)*

`Error saving credentials: error storing credentials - err: exit status 1, out: Cannot autolaunch D-Bus without X11 $DISPLAY`

... and the `login` does not succeed.

Many Linux systems do not have any sort of GUI environment installed on them, and so the `docker login` command should not assume that it may use any sort of "password safe" or "keyring" mechanism that exists only in such worlds.

**Comment by SundialServices** [comment]
P.S.  It doesn't matter if I do or don't use `sudo` and yes I am a member of the `docker` group.  Fundamentally, the command appears to want to rely on a "keyring" that doesn't exist outside of X11!  This is a very fundamental design flaw.

**Comment by thaJeztah** [comment]
Sounds like you may have the `golang-docker-credential-helpers` package installed; that package is not shipping with docker by default, but something that's packaged by Ubuntu (and possibly other distros). The `golang-docker-credential-helpers` provides helpers for both the `secretservice` and `pass` credential stores, but if `pass` is not installed, it picks `secretservice` as store, which is part of Gnome, and only works in interactive sessions.

**Comment by stefaanMLB** [comment]
@thaJeztah, so the solution is ... ?
Remove the `golang-docker-credential-helpers` or install `pass` ?

**Comment by thaJeztah** [comment]
Yes, if you want to run without X11 / run headless, then that's the solution

**Comment by kullervo610** [comment]
For me just installing 'pass' solved this issue.

**Comment by XanderStrike** [comment]
Installing `pass` worked for me, but I'm still getting the following message:

```
WARNING! Your password will be stored unencrypted in /home/xander/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store
```

I'm having trouble understanding why there's a hard, undocumented dependency on installing a password manager when my credentials are then stored in plaintext without using the password manager.

**Comment by robwold** [comment]
> Installing `pass` worked for me, but I'm still getting the following message:
> 
> ```
> WARNING! Your password will be stored unencrypted in /home/xander/.docker/config.json.
> Configure a credential helper to remove this warning. See
> https://docs.docker.com/engine/reference/commandline/login/#credentials-store
> ```
> 
> I'm having trouble understanding why there's a hard, undocumented dependency on installing a password manager when my credentials are then stored in plaintext without using the password manager.

@XanderStrike After installing pass there's more configuration:

1. Make sure `pass` is configured correctly, see e.g. https://medium.com/@thoferon/sharing-passwords-with-git-gpg-and-pass-628c2db2a9de
2. Set up docker to use pass as a store: https://github.com/docker/docker-credential-helpers

**Comment by drwetter** [comment]
pass on Debian has dependencies to X11 libs and other junk I don't want.

```
prompt:~ 0# apt-get install pass
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libice6 libqrencode4 libsm6 libxmu6 libxt6 qrencode tree x11-common xclip
Suggested packages:
  libxml-simple-perl
The following NEW packages will be installed:
  libice6 libqrencode4 libsm6 libxmu6 libxt6 pass qrencode tree x11-common xclip
0 upgraded, 10 newly installed, 0 to remove and 0 not upgraded.
Need to get 786 kB of archives.
After this operation, 1,669 kB of additional disk space will be used.
Do you want to continue? [Y/n] ^C
``` 

This is not a good solution for a server either.

**Comment by thaJeztah** [comment]
Have you tried with `--no-install-recommends` ? Looking at https://packages.debian.org/buster/pass, I see it has `gnupg` and `tree` as dependencies, other packages are recommended or suggested;

```bash
apt-get install --no-install-recommends pass
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  dirmngr gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm libassuan0 libksba8 libldap-2.4-2 libldap-common libnpth0 libreadline7 libsasl2-2 libsasl2-modules-db
  libsqlite3-0 lsb-base pinentry-curses readline-common tree
Suggested packages:
  dbus-user-session libpam-systemd pinentry-gnome3 tor parcimonie xloadimage scdaemon python python3 perl libxml-simple-perl ruby pinentry-doc readline-doc
Recommended packages:
  libsasl2-modules git qrencode xclip
The following NEW packages will be installed:
  dirmngr gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm libassuan0 libksba8 libldap-2.4-2 libldap-common libnpth0 libreadline7 libsasl2-2 libsasl2-modules-db
  libsqlite3-0 lsb-base pass pinentry-curses readline-common tree
0 upgraded, 24 newly installed, 0 to remove and 11 not upgraded.
Need to get 8558 kB of archives.
After this operation, 18.0 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```
